lib/render/mayaToHydra/CMakeLists.txt

```diff
@@ -1,36 +1,66 @@
 set(TARGET_NAME mtoh)
 
-set(SRC
-    defaultLightDelegate.cpp
-    plugin.cpp
-    pluginDebugCodes.cpp
-    renderGlobals.cpp
-    renderOverride.cpp
-    shadingModeExporter.cpp
-    tokens.cpp
-    utils.cpp
-    viewCommand.cpp
+add_library(${TARGET_NAME} SHARED)
+
+# -----------------------------------------------------------------------------
+# sources
+# -----------------------------------------------------------------------------
+target_sources(${TARGET_NAME} 
+    PRIVATE
+        defaultLightDelegate.cpp
+        plugin.cpp
+        pluginDebugCodes.cpp
+        renderGlobals.cpp
+        renderOverride.cpp
+        shadingModeExporter.cpp
+        tokens.cpp
+        utils.cpp
+        viewCommand.cpp
 )
 
-add_library(${TARGET_NAME} SHARED ${SRC})
+# -----------------------------------------------------------------------------
+# compile definitions
+# -----------------------------------------------------------------------------
+target_compile_definitions(${TARGET_NAME}
+    PRIVATE 
+        $<$<BOOL:${UFE_FOUND}>:WANT_UFE_BUILD>
+        # M3dView needs this (technically, MNativeWindowHdl.h)
+        # private atm because M3dView is only used in .cpp right now
+        # gcc will NOT set this automatically
+        $<$<BOOL:${IS_LINUX}>:LINUX>
+        # Not sure if msvcc sets this automatically, but won't hurt to redefine
+        $<$<BOOL:${IS_WINDOWS}>:_WIN32>
+)
 
-target_link_libraries(${TARGET_NAME} PRIVATE 
-    hd
-    hdMaya
-    hgi
-    usdShade
-    vt
-    ${MAYA_LIBRARIES}
+# -----------------------------------------------------------------------------
+# include directories
+# -----------------------------------------------------------------------------
+target_include_directories(${TARGET_NAME} 
+    PRIVATE 
+        "${CMAKE_CURRENT_SOURCE_DIR}/.."
```

Comment:
This relative pathing is kind of unfortunate. I'd rather see this use headers from the build or install tree.

Reply:
Totally agree. I am not really a big fan either. This was the way the cmake code came from mayaToHydra project and I wanted to discuss it with Paul as well.

Reply:
Yup, fine for now, but worth revisiting later.

Reply:
Agreed as well!
Actually, my hope is that once everything is properly modernized, we'll inherit this directory "for free", from linking against hdMaya, and we can drop this line entirely.